### ISSUE TITLE
mcs saving trials bug

### ISSUE THREAD

**Issue by ckingdon95** [anthofflab/Mimi.jl#308]
I haven't been able to figure out why yet, but the generate_trials function is saving one less trial than the number of trials specified (with calls to save_trial_inputs).

**Comment by rjplevin** [comment]
I'll look at it. Probably an off-by-one error owing to julia's 1-based indexing and my decades of programming in 0-based languages!

**Comment by ckingdon95** [comment]
out of curiosity, does the MCSIterator you defined get used by the save function?  if not, how does the `save(filename, mcs)` call on line 137 of montecarlo.jl work?

**Comment by rjplevin** [comment]
Yes, `save` uses the `IterableTable` protocol. `MCSIterator` provides the required methods.

**Comment by ckingdon95** [comment]
@rjplevin here's an example script that generates the bug. for any number of trials, one less value gets saved to "trials.csv". (and if you look at the rvdict, it has the correct number of values in the SampleStore, and it's the last value that doesn't get saved to the file)

```
using Mimi
using Distributions

trials = 3

mcs = @defmcs begin
    p1 = Normal()
end 

generate_trials!(mcs, trials; filename = "trials.csv")
```

**Comment by rjplevin** [comment]
Yeah, it was an off-by-one error after all:

`Base.done(mcs::MonteCarloSimulation, trialnum) = (trialnum == mcs.trials)` would be correct with 0-based indexing. It needs to be `(trialnum > mcs.trials)`.

I'm having some issues with my julia environment at the moment. Perhaps you can make this change?

**Comment by ckingdon95** [comment]
interesting, I though I had tried that change last week but let me try again!

**Comment by ckingdon95** [comment]
actually it appears to be just that line that needed to be changed. I'll add a test and open a PR with the change
